src/visualization/streamlit-exp-v2.py

```diff
@@ -0,0 +1,108 @@
+'''
+streamlit-exp-v2.py
+'''
+import numpy as np
+import pandas as pd
+import streamlit as st
+from pathlib import Path
+import seaborn as sns
+from bokeh.plotting import figure
+from bokeh.models import ColumnDataSource, CategoricalColorMapper
+
+
+def gen_random_data():
+    '''
+    Generates random data to represent a high-dimensional data set.
+    dimensions: (1000, 1001)
+    '''
+    np.random.seed(8)
+    # generate column names, data and label names
+    column_names = ['col_' + str(x) for x in range(1, 769)]
+    data = np.random.uniform(-10, 10, size=(1000, 768))
+    label_names = ['label_' + str(x) for x in range(1, 21)]
+
+    # insert numpy array, return pandas dataframe
+    df = pd.DataFrame(data, columns=column_names)
+
+    # generate list of labels and randomly insert into pandas dataframe
+    labels = np.random.choice(label_names, size=len(df))
+    df.insert(loc=0, column='label', value=labels)
+
+    return df
+
+
+@st.cache
+def gather_umap_data():
+    cwd = Path.cwd()
```

Comment:
In your source code, always give your functions docstrings. They can start with simple one line descriptions of what the function does. Later on you'll want to expand them into sphinx format and make them prettier.

Reply:
I added a docstring to the above. In my next branch, I'm expecting this to change, as this function's original purpose was to gather test data to ensure the streamlit/bokeh app produced a visualization.